### DISCUSSION TITLE
ALTER TABLE : ability to create more than one column in single query

### DISCUSSION THREAD

**Discussion by gdegentile1** [duckdb/duckdb#9676] [Ideas]
Hello,

I am currently evaluating DuckDB and I am already a big fan (I have previous experience with MonetDB which is unfortunately not very active).
I would like to create multiple columns in a single query.

**Example:**
`ALTER TABLE integers ADD COLUMN k INTEGER, n INTEGER;`

The reason is that I am processing OLAP cube Analytics which can rely on different in-memory databases (ex: H2, MonetDB, SQLITE, ...) and columns are created with a single query.
I have already adapted the code to create columns one by one in case the single query fails, but it would be nice to have this feature.

Thank you.

Best regards,
Guillaume

* OS: Windows
* DuckDB Client: Java
* DuckDB JDBC version: 0.9.1

**Comment by AhmedThahir** [comment]
Is this possible?

**Reply by datejada** [reply]
I don't think it is possible. For instance, I am trying something similar in my code (i.e., adding several columns in one query for efficiency purposes), but I get the following error message when trying to add multiple columns in one SQL query using DuckDB.

```log
ERROR: Parser Error: Only one ALTER command per statement is supported
```

DuckDB v1.2.1

**Comment by datejada** [comment]
@suvayu, with this limitation, any ideas to add several columns to a table in a connection using DuckDB?

Thanks!

**Reply by suvayu** [reply]
No idea, doesn't seem it's possible.  The [syntax](https://duckdb.org/docs/stable/sql/statements/alter_table#syntax) doesn't allow for multiple columns.

**Comment by gdegentile1** [comment]
Hi,

It is currently not supported by DuckDB, however it is common feature on
other DB (and probably SQL standard).

Can this feature be added to DuckDB?
In my case I am using a workaround approach to create columns one by one,
but it has an impact on performance.

Regards,
Guillaume

**Comment by alm818** [comment]
upvote this too, i need to add lots of columns
